ceph/rados/core_workflows.py

```diff
@@ -1034,18 +1035,238 @@ def get_journalctl_log(
+             host_object is osd host object
+             osd is the osd number
+        Returns:
+            PID number of the osd
+        """
+        try:
+            get_pid_cmd = (
+                'ps -ef | grep -w "/usr/bin/ceph-osd .* osd.'
+                + osd
+                + '"| grep -v "/run/podman-init" | grep -v grep | awk \'{print $2 }\''
+            )
+            osd_pid = host_object.exec_command(sudo=True, cmd=get_pid_cmd)
+            osd_pid = ("".join(osd_pid)).strip()
+            return osd_pid
+        except Exception as e:
+            log.error(f"Failed to  get the process id of OSD: {e}")
 
-        ceph config set osd_mclock_profile <profile_name>
+    def check_osd_status(self, host_object, osd):
+        """
+        Used to check the process is running in osd node or not
+        Args:
+             host_object is osd host object
+             osd is the osd number
+        Returns:
+            If process is up and running then  returns 0 else return non zero
+        """
+        try:
+            osd_status_cmd = (
+                'systemctl is-active --quiet "$(systemctl list-units --type=service | grep -i "ceph-.*@osd.'
+                + osd
+                + ".service\" | awk '{print $1}')\" && echo $?"
+            )
+            status = host_object.exec_command(
+                sudo=True, cmd=osd_status_cmd, long_running=True
+            )
+            if type(status) == tuple:
+                status = ("".join(status)).strip()
+            return status
+        except Exception as e:
+            log.error(f"Failed to check the OSD status:{e}")
 
-        profile names:
-        - balanced
-        - high_recovery_ops
-        - high_client_ops
+    def kill_osd_process(self, host_object, osd_pid):
+        """
+        Used to kill the osd process
+        Args:
+            host_object is osd host object
+            osd is the osd number
+        Returns:
+            If process is up and running then  returns 0 else return non zero
+        """
+        try:
+            osd_kill_cmd = "kill -9  " + osd_pid
+            kill_status = host_object.exec_command(sudo=True, cmd=osd_kill_cmd)
+            time.sleep(5)
+            return kill_status
+        except Exception as e:
+            log.error(f"Failed to kill the OSD process:{e}")
 
+    def flag_chk(self, flag):
+        """
+        To get the pg_id list from the cluster
         Args:
-            profile: mclock profile name
-            osd: "osd" service by default or "osd.<Id>"
+            flag: Rados flag to set
+            Returns : True -> pass, False -> fail
         """
-        return self.node.shell([f"ceph config set {osd} osd_mclock_profile {profile}"])
+        ceph_health_cmd = "ceph health --format json"
+        try:
+            health_dump = self.run_ceph_command(cmd=ceph_health_cmd)
+            health_dump_json = json.dumps(health_dump)
+            health_load_json = json.loads(health_dump_json)
```

Comment:
reduntant. `run_ceph_command` returns out dict.

Reply:
Removed the lines.